Source/Tools/FEXServer/CoreDumpService.h

```diff
@@ -0,0 +1,202 @@
+#pragma once
+#include "Common/FEXServerClient.h"
+#include "Unwind/FileMapping.h"
+
+#include <FEXCore/fextl/map.h>
+#include <FEXCore/fextl/list.h>
+#include <FEXCore/fextl/string.h>
+#include <FEXCore/fextl/unordered_map.h>
+#include <FEXCore/fextl/vector.h>
+
+#include <cstring>
+#include <poll.h>
+#include <sys/socket.h>
+#include <thread>
+
+namespace Unwind {
+class Unwinder;
+}
+
+namespace CoreDumpService {
+  class CoreDumpClass {
+    public:
+      struct Description {
+        uint32_t pid;
+        uint32_t tid;
+        uint32_t uid;
+        uint32_t gid;
+        uint32_t Signal;
+        uint64_t Timestamp;
+        uint8_t HostArch;
+        uint8_t GuestArch;
+      };
+
+
+      int Init() {
+        int SVs[2];
+        int Result = socketpair(AF_UNIX, SOCK_STREAM | SOCK_CLOEXEC, 0, SVs);
```

Comment:
How does `socketpair` compare against `pipe2`? Since we use both, how are choosing between the two?

Reply:
We primarily use Sockets here because we need to pass FDs through sendmsg. pipe doesn't give you that ability.
We actually use pipe in fairly limited number of locations. Pipe is primarily used for logging.

The only other place pipe is used is if `pidfd_open` fails, it will fallback to a pipe fd instead. Which is only used for `FEXServer -w`.